final_task/pycalc/pycalc.py

```diff
@@ -0,0 +1,190 @@
+"""This is a main module with general functions
+
+:author: Fyodorova Elena
+"""
+from re import findall
+import pycalc.Data as Data
+import argparse
+import importlib
+
+
+def pycalc(string: str):
+    """Main function
+
+    :param string: input expression
+    :type string: str
+    :return: result of calculation or error message
+    """
+    brackets_list = findall('[(-)]', string)  # Searching for brackets to check a balance of them
+    if brackets_list.count('(') != brackets_list.count(')'):
+        return 'ERROR: brackets are not balanced'
+    for function in findall(
+            '[a-zA-Z]+|[a-zA-Z]+\\d+|\\d+[a-zA-Z]+',
+            string):
+        if function not in Data.FUNCTION_DICT:
+            return "ERROR: '{0}' is unknown function or constant".format(function)
+    # Searching for unknown number and letter combinations
+    if findall('[0-9.]+', string) == [string]:
```

Comment:
This regular expression will return `True` even for input string like `string="123.2345.345...8"` that will cause ValueError upon casting this `string` into `float()`